src/main/java/com/sourcery/gymapp/backend/kafka/streams/NotificationsLikesAggregator.java

```diff
@@ -0,0 +1,98 @@
+package com.sourcery.gymapp.backend.kafka.streams;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.sourcery.gymapp.backend.events.LikeNotificationEvent;
+import com.sourcery.gymapp.backend.events.RoutineLikeEvent;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.Grouped;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.support.serializer.JsonSerde;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.util.UUID;
+
+@Slf4j
+@Component
+public class NotificationsLikesAggregator {
+    @Value("${spring.kafka.topics.likes-events}")
+    private String INPUT_TOPIC;
+    @Value("${spring.kafka.topics.likes-notifications}")
+    private String OUTPUT_TOPIC;
+    @Value("${spring.kafka.aggregation.like-notification.interval-minutes}")
+    private int aggregationInterval;
+
+    private final ObjectMapper objectMapper;
+
+    public NotificationsLikesAggregator() {
+        this.objectMapper = new ObjectMapper();
+        this.objectMapper.registerModule(new JavaTimeModule());
+    }
+
+
+    @Bean
+    public KStream<String, LikeNotificationEvent> notificationLikesStream(StreamsBuilder builder) {
```

Comment:
The NotificationsLikesAggregator lacks event volume limits within time windows, risking performance issues if receiving >10,000 likes/minute. Consider implementing throttling or batch processing to handle traffic spikes.

(I would not do anything right now with this, its just FYI)